scheduler/src/cook/rest/api.clj

```diff
@@ -918,13 +916,19 @@
                 (assoc :partitions (into #{} (walk/stringify-keys partitions))))))
        (into #{})))
 
-(defn validate-job-gpu-model
-  "Validates that GPU model entered is supported on the pool"
-  [pool-name {:keys [env]}]
-  (let [requested-gpu-model (get env "COOK_GPU_MODEL")]
+(defn validate-gpu-job
+  "Validates that a job requesting GPUs is supported on the pool"
+  [gpu-enabled? pool-name {:keys [gpus env]}]
+  (let [requested-gpu-model (get env "COOK_GPU_MODEL")
+        gpus' (or gpus 0)]
+    (when (and (pos? gpus') (not gpu-enabled?))
+      (throw (ex-info (str "GPU support is not enabled") {})))
     (when (and requested-gpu-model
                (not (contains? (get-gpu-models-on-pool (config/valid-gpu-models) pool-name) requested-gpu-model)))
-      (throw (ex-info (str "The following GPU model is not supported: " requested-gpu-model) {})))))
+      (throw (ex-info (str "The following GPU model is not supported: " requested-gpu-model) {})))
+    (when (and (and (pos? gpus') (not requested-gpu-model))
```

Comment:
Why is `(not requested-gpu-model)` part of this check? Shouldn't this error even if I did request a specific GPU model but I asked for it on a Cook pool that has no GPU models config?

Reply:
Good point. In that case I will move this check above the previous one because I think it is broader than the one above.